### PR TITLE
feature/timeline-filter-for-transactions

### PR DESCRIPTION
**Pull request by predrag-babic**

### CODE REVIEW

File: components/TimelineFilter.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        <StyledTimelineFilterButton
+          key={timeframe.value}
+          $active={selectedTimeframe === timeframe.value}
+          onClick={() => handleTimeframeClick(timeframe.value)}
+        >
+          {timeframe.label}
+        </StyledTimelineFilterButton>
+      ))}
+    </StyledTimelineFilterContainer>
+  );
```
Comment (F-Kirchhoff) [inline components/TimelineFilter.js:33]: this is great code ✨

File: components/TimelineFilter.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export default function TimelineFilter({
+  selectedTimeframe,
+  onTimeframeChange,
+}) {
+  const timeframes = [
+    { label: "7 Days", value: 7 },
+    { label: "30 Days", value: 30 },
+    { label: "Half Year", value: 180 },
+    { label: "Full Year", value: 365 },
+  ];
```
Comment (F-Kirchhoff) [inline components/TimelineFilter.js:12]: not really an issue, but this data never changes but is recreated with every render. if you move it out of the function body it is only generated once and reused for every render.

File: components/TimelineFilter.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  function handleTimeframeClick(value) {
+    if (selectedTimeframe === value) {
+      onTimeframeChange(null);
+    } else {
+      onTimeframeChange(value);
+    }
+  }
```
Comment (F-Kirchhoff) [inline components/TimelineFilter.js:20]: you could move this function to the index.js and replace the function 'handleTimeFrameChange' with it.

File: pages/index.js [hunk 4]
```diff
@@ -217,27 +234,38 @@ export default function HomePage({
...
+          onCloseModal={closeModal}
+          handleOpenDeleteDialogue={handleOpenDeleteDialogue}
+          handleConfirmDelete={handleConfirmDelete}
+          handleCancelDeleteDialogue={handleCancelDeleteDialogue}
+          handleDeleteTransaction={handleDeleteTransaction}
+          isDeletingId={isDeletingId}
+          sortOrder={sortOrder}
+        />
+      ) : (
+        <p>No transactions found within this timeframe.</p>
```
Comment (F-Kirchhoff) [inline pages/index.js:267]: there are other possiblities for the transaction list being empty. Please adapt the text such that it more general like "No transactions found".

**Review by F-Kirchhoff — APPROVED** [review]
Hey Money Bin Team,
this is a fantastic PR, good job. There are optional optimizations that you could do, I listed them below.

File: components/TimelineFilter.js (hunk 3 above)
Comment (predrag-babic) [reply components/TimelineFilter.js]: Hi,
nice hint, thanks.
i moved it to the index.js and replace it in the TimelineFilter component.

**Pull request merged**